plugins/org.jnario.lib/src/org/jnario/lib/Should.java

```diff
@@ -174,4 +223,610 @@ public void describeTo(Description description) {
+		return false;
+	}
+
+	private static boolean isJnarioLink(String url, Object source) {
+		if (source == null || url == null) {
+			return false;
+		}
+		//
+		try {
+			String ref = url;
+			if (ref.startsWith("#")) { //$NON-NLS-1$
+				ref = "./" + source.getClass().getSimpleName() + ".html" + ref; //$NON-NLS-1$ //$NON-NLS-2$
+			}
+
+			String baseRef;
+			String[] fragments = null;
+			if (ref.contains(".html#")) { //$NON-NLS-1$
+				String[] parts = ref.split(java.util.regex.Matcher.quoteReplacement(".html#")); //$NON-NLS-1$
+				if (2 != parts.length) {
+					return false;
+				}
+				fragments = parts[1].split(java.util.regex.Matcher.quoteReplacement("_") + "+"); //$NON-NLS-1$ //$NON-NLS-2$
+				StringBuilder b = new StringBuilder();
+				for (String s : fragments) {
+					if (s.length() > 1) {
+						b.append(s.substring(0, 1).toUpperCase() + s.substring(1));
+					} else {
+						b.append(s.toUpperCase());
+					}
+				}
+				if (parts[0].endsWith("Spec")) { //$NON-NLS-1$
+					baseRef = parts[0] + ".html";
+					ref = parts[0].substring(0, parts[0].length() - 4)
+							+ b.toString() + "Spec.html"; //$NON-NLS-1$
+				} else {
+					baseRef = parts[0] + "Spec.html";
+					ref = parts[0] + b.toString() + "Spec.html"; //$NON-NLS-1$
+				}
+			} else {
+				baseRef = source.getClass().getName().replaceAll("\\.", "/") + ".html";
+			}
+			//
+			Class<?> jnarioSpec = getJnarioSpec(source.getClass(), ref);
+			if (jnarioSpec == null) {
+
+				// The specification could be a function of the Java class.
+				if (fragments != null) {
+					jnarioSpec = getJnarioSpec(source.getClass(), baseRef);
+					if (jnarioSpec != null) {
+						StringBuilder operationName = new StringBuilder();
+						for (String fragment : fragments) {
+							if (operationName.length() > 0) {
+								operationName.append(fragment.substring(0, 1).toUpperCase() + fragment.substring(1).toLowerCase());
+							} else {
+								operationName.append(fragment.toLowerCase());
+							}
+						}
+						String operationNameStr = "_" + operationName.toString(); //$NON-NLS-1$
+						try {
+							jnarioSpec.getDeclaredMethod(operationNameStr);
+							return true;
+						} catch (Throwable _) {
+							// Failure
+						}
+					}
+				}
+
+				return false;
+			}
+			return true;
+		} catch (Throwable _) {
+			return false;
+		}
+	}
+
+	/** Replies the type that corresponds to the specified class.
+	 * If the name corresponds to a primitive type, the low-level type
+	 * will be replied.
+	 * This method extends
+	 * {@link Class#forName(String)} with autoboxing support.
+	 * 
+	 * Copied from {@link "https://github.com/gallandarakhneorg/afc/blob/master/core/vmutils/src/main/java/org/arakhne/afc/vmutil/ReflectionUtil.java"}.
+	 * 
+	 * @param name is the name of the class to load.
+	 * @return the loaded class
+	 * @throws ClassNotFoundException  if name names an
+	 * unknown class or primitive
+	 */
+	private static Class<?> forName(String name) throws ClassNotFoundException {

```

Comment:
The if ``$string.equals(name)``s here look like a good candidate for Xtend's switch. Is there anything holding this back from being an Xtend file?
```xtend
switch name {
    case null, case '', case 'null' : void.class
    case 'boolean' : boolean.class
    case 'byte' : byte.class
    ...
    default : Class.forName(name)
}
```